### PR TITLE
generate an basic indexer test for Valkyrie::Resource works

### PR DESCRIPTION
**Pull request by elrayle**
As part of the work resource generation process, this adds a spec test for the generated indexer.

@samvera/hyrax-code-reviewers

### CODE REVIEW

**Comment by elrayle** [comment]
The test failure is the same for all builds.  The same test failure does not occur when run by itself locally.  The failure does not appear to be related to this PR.

Failing test:
```
rspec ./spec/features/collection_type_spec.rb:367 # collection_type delete collection type when collections exist of this type shows unable to delete dialog and forwards to All Collections with filter applied
```

Failure:
```
Failure/Error: expect(page).to have_content(deny_delete_modal_text, :all)
     ArgumentError:
       ${@type} is not a valid type for a text query
```

File: lib/generators/hyrax/work_resource/templates/indexer_spec.rb.erb [hunk 1]
```diff
@@ -0,0 +1,16 @@
+# frozen_string_literal: true
+
+# Generated via
+#  `rails generate hyrax:work_resource <%= class_name %>`
+require 'rails_helper'
+require 'hyrax/specs/shared_specs/indexers'
+
+RSpec.describe <%= class_name %>Indexer do
+  let(:indexer_class) do
```
Comment (no-reply) [inline lib/generators/hyrax/work_resource/templates/indexer_spec.rb.erb:9]: doesn't this test the wrong class?

the `describe` says this tests `[GeneratedModel]Indexer`, is there a reason we'd want to test a generic `Hyrax::ResourceIndexer` instead?

**Comment by no-reply** [comment]
looks like the tests are fixed in #4409

File: lib/generators/hyrax/work_resource/templates/indexer_spec.rb.erb (hunk 1 above)
Comment (elrayle) [reply lib/generators/hyrax/work_resource/templates/indexer_spec.rb.erb:9]: That was likely a copy/paste error.  I'll take a look.

Comment (elrayle) [reply lib/generators/hyrax/work_resource/templates/indexer_spec.rb.erb]: @no-reply Updated to correctly use the generated index class.

File: lib/generators/hyrax/work_resource/templates/indexer_spec.rb.erb [hunk 2]
```diff
@@ -0,0 +1,12 @@
+# frozen_string_literal: true
+
+# Generated via
+#  `rails generate hyrax:work_resource <%= class_name %>`
+require 'rails_helper'
+require 'hyrax/specs/shared_specs/indexers'
+
+RSpec.describe <%= class_name %>Indexer do
+  let(:indexer_class) { <%= class_name %>Indexer }
```
Comment (no-reply) [inline lib/generators/hyrax/work_resource/templates/indexer_spec.rb.erb:9]: would `described_class` be better?

```suggestion
  let(:indexer_class) { described_class }
```

Comment (elrayle) [reply lib/generators/hyrax/work_resource/templates/indexer_spec.rb.erb:9]: That makes sense.

Comment (elrayle) [reply lib/generators/hyrax/work_resource/templates/indexer_spec.rb.erb]: Done

Comment (no-reply) [reply lib/generators/hyrax/work_resource/templates/indexer_spec.rb.erb]: thanks!

**Comment by no-reply** [comment]
thanks for this!

**Pull request merged**